src/submap.h

```diff
@@ -418,6 +418,35 @@ class maptile_impl
             return *std::prev( sm->get_items( pos() ).cend() );
         }
 
+        /** Returns the uppermost favorite item if there is a favorite item. Otherwise, returns the item with the greatest volume.
+        *
+        * Assumes there is at least one item.
+        * Used to decide which item to render.
+        * TODO: might be slow enough to justify storing the most visible item for each tile? 
+        */
+        const item &get_most_visible_item() const {
```

Comment:
The implementation of this size should be in the `.cpp` file, not the `.h` file.

Btw this will improve the compilation time when changing only the implementation enormously, as you only recompile one file.

Reply:
I realized that - I was again just basing it off what get_uppermost_item() did, and assumed that there was some reason that other functions in the same file were implemented in the header. I'll fix that.

Reply:
It's based on simplicity of the function. One liners can be in .h files. But anything in .h means drastically more recompile time on change.

Reply:
oh wait, actually it's all in a header because it's a method of a template class.

Reply:
Oh, you might be right. I don't know enough about template stuff splitting, Feel free to mark this as resolved.